lib/easypost/client.rb

```diff
@@ -89,7 +87,7 @@ def make_request(
     potential_error = EasyPost::Errors::ApiError.handle_api_error(response)
     raise potential_error unless potential_error.nil?
 
-    EasyPost::InternalUtilities::Json.convert_json_to_object(response.body, cls)
+    EasyPost::InternalUtilities::Json.parse_json(response.body)
```

Comment:
Reviewer's note:

Instead of directly converting the response into an EasyPost object, just parse it and let the services do the conversion. This allows us to have an intermediate state (a `Hash`) so then the services inject the filters and shift the conversion to EasyPost Models responsibility to services.

Reply:
This improvement could have been on a separate PR so reviewers only focus on the changes of the actual PR title which is the pagination redesign.

Reply:
Although that's the easy solution, problem is that we'd need to pass `filters` to the client and do the injection of `filters` before the conversion to `EasyPostObject` happens, which is one too many responsibilities for the client. Hence why I went with the client only parsing the JSON response to the applicable Ruby object with the added benefit of improving maintainability and clearer separation of concerns instead of keep kicking the can down the road.

Reply:
I was suggesting having this improvement on a separate PR so the reviewer can narrow down the actual changes for pagination redesign